src/rime/config/config_compiler.cc

```diff
@@ -155,17 +155,22 @@ inline static string StripOperator(const string& key, bool adding) {
 }
 
 // defined in config_data.cc
-bool TraverseCopyOnWrite(an<ConfigItemRef> root, const string& path,
-                         function<bool (an<ConfigItemRef> target)> writer);
+an<ConfigItemRef> TypeCheckedCopyOnWrite(an<ConfigItemRef> parent,
+                                         const string& key);
+an<ConfigItemRef> TraverseCopyOnWrite(an<ConfigItemRef> root,
+                                      const string& path);
 
 static bool EditNode(an<ConfigItemRef> target,
```

Comment:
The name `target` is used for another (argument) variable in lambda, which can refer to a different config node. Better differentiate the name.